File: crates/weave-graph-parse/tests/fixtures/sample.cs
using System;

namespace App {
    interface INamed {
        string GetName();
    }

    class Greeter : INamed {
        public string GetName() {
            return this.name;
        }

        public string Greet() {
            return FormatName(this.GetName());
        }
    }
}

File: crates/weave-graph-parse/tests/fixtures/sample.cpp
#include <string>

int add(int a, int b) {
    return a + b;
}

namespace app {

struct Point {
    int x;
    int y;
};

class Base {
public:
    virtual void name();
};

class Greeter : public Base {
public:
    std::string greet() {
        Base::name();
        return this->helper();
    }
    std::string helper() {
        return formatName();
    }
    int* getPtr() {
        return nullptr;
    }
};

}

struct {
    int anon_field;
};

namespace {
    void secret() {}
}
